### ISSUE TITLE
Editor loses focus when selecting from the special-characters dropdown.

### ISSUE THREAD

**Issue by mateuszzagorski** [ckeditor/ckeditor5#12121]
## 📝 Provide detailed reproduction steps (if any)

1.  Start the docs server.
2. Open special characters feature guide: http://127.0.0.1:8080/ckeditor5/latest/features/special-characters.html#demo.
3. Focus something inside the editor and focus toolbar (option/alt+f10).
4. Go to the special-characters button. Press arrow_down, select anything from the dropdown.

### ✔️ Expected result

Focus is back on the dropdown.

### ❌ Actual result

https://user-images.githubusercontent.com/11699569/180229022-6d69c740-f04f-4a86-8f0d-a25059afaade.mp4

## 📃 Other details

This cannot be replicated in https://ckeditor.com/docs/ckeditor5/latest/features/special-characters.html#demo.
The document.activeElement after the provided steps is `<body>`.
Same outcome can be achieved when using mouse.

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by mlewand** [comment]
This results with special characters disappearing with inline toolbar.

**Comment by mlewand** [comment]
Closely related to #12125, see https://github.com/ckeditor/ckeditor5/issues/12125#issuecomment-1192327029.
